pkg/compact/compact.go

```diff
@@ -342,12 +347,12 @@ func (c *Syncer) removeIfMetaMalformed(ctx context.Context, id ulid.ULID) (remov
 
 // GroupKey returns a unique identifier for the group the block belongs to. It considers
 // the downsampling resolution and the block's labels.
-func GroupKey(meta metadata.Meta) string {
-	return groupKey(meta.Thanos.Downsample.Resolution, labels.FromMap(meta.Thanos.Labels))
+func GroupKey(meta metadata.Thanos) string {
+	return groupKey(meta.Downsample.Resolution, labels.FromMap(meta.Labels))
 }
 
 func groupKey(res int64, lbls labels.Labels) string {
-	return fmt.Sprintf("%d@%s", res, lbls)
+	return fmt.Sprintf("%d@%s", res, fmt.Sprintf("%v", lbls.Hash()))
```

Comment:
Why two nested `Sprintf` calls?

```suggestion
	return fmt.Sprintf("%d@%d", res, lbls.Hash())
```

Reply:
:man_facepalming: